src/BenchmarkDotNet/Portability/RuntimeInformation.cs

```diff
@@ -173,10 +168,21 @@ private static string GetNetCoreVersion()
             }
             else
             {
-                var coreclrAssemblyInfo = FileVersionInfo.GetVersionInfo(typeof(object).GetTypeInfo().Assembly.Location);
-                var corefxAssemblyInfo = FileVersionInfo.GetVersionInfo(typeof(Regex).GetTypeInfo().Assembly.Location);
+                string coreclrLocation = typeof(object).GetTypeInfo().Assembly.Location;
+                string corefxLocation = typeof(Regex).GetTypeInfo().Assembly.Location;
 
-                if (CoreRuntime.TryGetVersion(out var version) && version >= new Version(5, 0))
+                // Handle cases where assembly location is empty (e.g. single-file publish, AOT, some test runners)
+                if (string.IsNullOrEmpty(coreclrLocation) || string.IsNullOrEmpty(corefxLocation))
+                {
+                    return CoreRuntime.TryGetVersion(out var ver) && ver.Major >= 5
+                        ? $".NET {ver} (assembly location unavailable)"
+                        : $".NET Core {ver?.ToString() ?? Unknown} (assembly location unavailable)";
+                }
+
+                var coreclrAssemblyInfo = FileVersionInfo.GetVersionInfo(coreclrLocation);
+                var corefxAssemblyInfo = FileVersionInfo.GetVersionInfo(corefxLocation);
+
+                if (CoreRuntime.TryGetVersion(out var version) && version.Major >= 5)
```

Comment:
.NET Core 3.x is out support for several years. Should these special cases be dropped to make the logic simpler?

Reply:
I meant just drop this special code that tries to differentiate between coreclr and corefx versions.